common/src/OpenSpaceNet.cpp

```diff
@@ -363,359 +400,282 @@ void OpenSpaceNet::initFilter()
+{
+    auto slidingWindow = dg::deepcore::imagery::node::SlidingWindow::create("SlidingWindow");
+    auto windowSizes = calcWindows();
+    auto resampledSize = args_.resampledSize ?  
+                         cv::Size {*args_.resampledSize, (int) roundf(modelAspectRatio_ * (*args_.resampledSize))} : 
+                         cv::Size {};
+    slidingWindow->attr("windowSizes") = windowSizes;
+    slidingWindow->attr("resampledSize") = resampledSize;
 
-        if(!skip) {
-            addFeature(prediction);
-        }
-    }
+    return slidingWindow;
 }
 
-void OpenSpaceNet::processSerialBoxes()
+LabelFilter::Ptr OpenSpaceNet::initLabelFilter(bool isSegmentation)
 {
-    auto predictions = processSerial<PredictionBox>();
-
-    if(args_.action != Action::LANDCOVER && args_.nms) {
-        skipLine();
-        OSN_LOG(info) << "Performing non-maximum suppression..." ;
-        predictions = nonMaxSuppression(predictions, args_.overlap / 100);
+    LabelFilter::Ptr labelFilter;
+    if (isSegmentation) {
+        labelFilter = PolyLabelFilter::create("LabelFilter");
+    } else {
+        labelFilter = BoxLabelFilter::create("LabelFilter");
     }
 
-    OSN_LOG(info) << predictions.size() << " features detected.";
-
-    for(const auto& prediction : predictions) {
-        addFeature((PredictionPoly)prediction);
+    if(!args_.excludeLabels.empty()) {
+        labelFilter->attr("labels") = vector<string>(args_.excludeLabels.begin(),
+                                                     args_.excludeLabels.end());
+        labelFilter->attr("labelFilterType") = LabelFilterType::EXCLUDE;
+    } else if(!args_.includeLabels.empty()) {
+        labelFilter->attr("labels") = vector<string>(args_.includeLabels.begin(),
+                                                     args_.includeLabels.end());
+        labelFilter->attr("labelFilterType") = LabelFilterType::INCLUDE;
+    } else {
+        return nullptr;
     }
+
+    return labelFilter;
 }
 
-template<class T>
-std::vector<T> OpenSpaceNet::processSerial()
+PredictionToFeature::Ptr OpenSpaceNet::initPredictionToFeature()
 {
-    skipLine();
-    OSN_LOG(info) << "Processing...";
+    auto predictionToFeature = PredictionToFeature::create("PredictionToFeature");
+    predictionToFeature->attr("geometryType") = args_.geometryType;
+    predictionToFeature->attr("pixelToProj") = pixelToProj_;
+    predictionToFeature->attr("topNName") = "top_five";
+    predictionToFeature->attr("topNCategories") = 5;
 
-    startProgressDisplay();
+    return predictionToFeature;
+}
 
-    auto startTime = high_resolution_clock::now();
-    auto mat = GeoImage::readImage(*image_, bbox_, regionFilter_.get(), [this](float progress) -> bool {
-        if(pd_ && !args_.quiet) {
-            pd_->update("Reading", progress);
+WfsFeatureFieldExtractor::Ptr OpenSpaceNet::initWfs()
+{
+    if (args_.dgcsCatalogID || args_.evwhsCatalogID) {
```

Comment:
All this stuff should probably be moved to DeepCore.

Reply:
concur

Reply:
Do you mean that all of the stuff that goes into initializing the WFS should go into deepcore? Currently the WFS stuff is highly generalized, is the thought that there should be subclasses to those nodes like "DGCSFeatureFieldExtractor" etc?